apps/coordinator/src/components/Wallet/AddressExpander.jsx

```diff
@@ -1,454 +1,99 @@
-              </Box>
-            )}
-            {interactionMessage === "" && interactionError === "" && (
-              <InteractionMessages
-                messages={this.interaction.messagesFor({
-                  state: interactionState,
-                })}
-              />
-            )}
-            <Button
-              variant="contained"
-              size="large"
-              onClick={this.confirmOnDevice}
-              disabled={interactionState === ACTIVE}
-            >
-              Confirm
-            </Button>
-            {(interactionMessage !== "" || interactionError !== "") && (
-              <Button size="large" onClick={this.resetInteractionState}>
-                Reset
-              </Button>
-            )}
-          </>
-        )}
-      </Grid>
-    );
-  };
-
-  resetInteractionState = () => {
-    this.setState({
-      interactionError: "",
-      interactionMessage: "",
-      interactionState: PENDING,
-    });
-  };
-
-  confirmOnDevice = async () => {
-    this.setState({ interactionState: ACTIVE });
-    const { node } = this.props;
-    const { multisig } = node;
-    try {
-      const confirmed = await this.interaction.run();
-      if (
-        confirmed.address === multisig.address &&
-        confirmed.serializedPath === this.interaction.bip32Path
-      ) {
-        this.setState({
-          interactionState: ACTIVE,
-          interactionMessage: "Success",
-          interactionError: "",
-        });
-      } else {
-        this.setState({
-          interactionState: ACTIVE,
-          interactionError: "An unknow error occured",
-          interactionMessage: "",
-        });
-      }
-    } catch (error) {
-      this.setState({
-        interactionState: ACTIVE,
-        interactionError: error.message,
-        interactionMessage: "",
-      });
-    }
-  };
-
-  keySelected = (event, extendedPublicKeyImporter) => {
-    const { network, node } = this.props;
-    const { multisig, bip32Path } = node;
-
-    this.interaction = ConfirmMultisigAddress({
-      keystore: extendedPublicKeyImporter.method,
-      network,
-      bip32Path: `${extendedPublicKeyImporter.bip32Path}${bip32Path.slice(1)}`,
-      multisig,
-    });
-    this.setState({ hasInteraction: true });
-    this.resetInteractionState();
-  };
-
-  // TODO: DRY out with test
-  confirmAddressDescription() {
-    const { network, node, requiredSigners, totalSigners } = this.props;
-    const { multisig } = node;
-    const addressType = multisigAddressType(multisig);
-
-    return (
-      <Box>
-        <p>
-          Confirm the following {network} {addressType} {requiredSigners}
-          -of-
-          {totalSigners} multisig address on your device:
-        </p>
 
+/**
+ * AddressExpander displays details for a single wallet address (node),
+ * including its BIP32 path, balance, UTXO count, and a table of UTXOs.
+ * Used as an expandable panel in the wallet UI.
+ */
+function AddressExpander(props) {
```

Comment:
Since you're making this a whole new functional component anyway (which is great!), you can now just use `useSelector` to pull from the redux store without connect. It would also be nice to convert to typescript when we have the opportunity.